llvm/utils/repo/Makefile

```diff
@@ -56,20 +64,18 @@ libcompiler-rt.a:
 		-D CMAKE_TOOLCHAIN_FILE=$(TOOLCHAIN_FILE)                     \
 		-D CMAKE_CXX_COMPILER=$(LLVM_BUILD)/bin/clang++               \
 		-D CMAKE_C_COMPILER=$(LLVM_BUILD)/bin/clang                   \
+		-D utils_dir=$(LLVM_REPO)/llvm/utils/repo                     \
 		-D CMAKE_INSTALL_PREFIX=$(LLVM)                               \
 		-D musl_install=$(MUSL)                                       \
 		../compiler-rt &&                                             \
 	make -j 8 &&                                                          \
 	make install
 
-$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
-$(LLVM)/lib/linux/clang_rt.crtend-x86_64.o: libcompiler-rt.a
+$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o.elf: libcompiler-rt.a
+	repo2obj --repo=$(LLVM_REPO)/build_libcompiler-rt/libcompiler-rt.db $(LLVM_REPO)/build_libcompiler-rt/lib/linux/clang_rt.crtbegin-x86_64.o -o $@

```

Comment:
I would break this up over multiple lines as you have done above to make it easier to read.

I’m confused by the use of libcompiler-rt.a as the prerequisite of this rule but where it doesn’t appear in the command that is executed. Shouldn’t the prerequisite be clang_rt.crtbegin-x86_64.o?

(This also applies to the next diff, of course.)

Reply:
Fixed in the commit https://github.com/SNSystems/llvm-project-prepo/pull/155/commits/ce1d66d51a4d69b1046d9f2abdb013408110f63b.